### PR TITLE
TSNE vis: update the model & embeddings

### PR DESCRIPTION
**Pull request by bzz**
Some improvements in visualisation relevant to #62.

It's using a ['all-MiniLM-L6-v2' model](https://www.sbert.net/docs/pretrained_models.html) (87Mb instead of 418Mb, the same 512 context size) that is faster & seems to provide better visualisation.

**Before**: the previous model 
Screenshot 2023-04-30 at 18 24 44

The previous model using both, abstracts and titles.
Screenshot 2023-04-30 at 18 25 31

The new model (batch size 1, abstracts only)
Screenshot 2023-04-30 at 18 26 51

**After**: the new model + titles + batched 
Screenshot 2023-04-30 at 18 25 04

I tied UMAP for it and the results seems less interesting (but didn't experiment much)

<details>
 <summary>UMAP</summary>

Screenshot 2023-04-30 at 18 54 17

</details>

I also tried using a larger 420Mb model fine-tuned on scientific papers from [SciRepEval](https://api.semanticscholar.org/CorpusID:254018137) - [allenai/specter2](https://huggingface.co/allenai/specter2) \w [proximity adaptor](https://huggingface.co/allenai/specter2_proximity) that takes 1.5min vs 30sec of the above. It can't be switched though the CLI only, as it requires loading an adaptor. 

<details>
 <summary>specter2 TSNE </summary>

Screenshot 2023-04-30 at 18 43 35

</details>

Let me know what you think and which one do you prefer!

### CODE REVIEW

**Comment by mallamanis** [comment]
Thanks for looking into this @bzz! All these options seem quite interesting, yet, it's hard to decide without looking at what each point represents.

May I suggest, that you sample 3-4 papers that you know and try to see which of the visualizations gets "reasonable" neighbors? I'd be happy to go with whatever you find more useful, in that sense 👓

**Comment by bzz** [comment]
Indeed, that was exactly what I did, I apologize for missing this crucial information :) 

Here are the interactive visualisation of embeddings + metadata for all 4 different models, sorted by subjective perception of "cluster quality" that is easy to explore:
 * [T-SNE GPT-3](https://projector.tensorflow.org/?config=https://gist.githubusercontent.com/bzz/a4481b985e690dc937749156f7b851c2/raw/2c39708836065e8d75c479d7802dee0eddc54fe6/GPT-3-projector_config.json), [model](https://openai.com/blog/new-and-improved-embedding-model) (text-embedding-ada-002)
 * [T-SNE sentence-transformers/all-MiniLM-L6-v2](https://projector.tensorflow.org/?config=https://gist.githubusercontent.com/bzz/a4481b985e690dc937749156f7b851c2/raw/2c39708836065e8d75c479d7802dee0eddc54fe6/all-MiniLM-L6-v2-projector_config.json), [model](https://huggingface.co/sentence-transformers/all-MiniLM-L6-v2) (new model + titles + batched)
 * [T-SNE allenai/specter2](https://projector.tensorflow.org/?config=https://gist.githubusercontent.com/bzz/a4481b985e690dc937749156f7b851c2/raw/2c39708836065e8d75c479d7802dee0eddc54fe6/specter2-projector_config.json), [model](https://huggingface.co/allenai/specter) (specter2)
 * [T-SNE deepset/sentence_bert](https://projector.tensorflow.org/?config=https://gist.githubusercontent.com/bzz/a4481b985e690dc937749156f7b851c2/raw/15c31f3404b6d8a5d6c23cb4d9e1978e35e0a2a2/sentence_bert-projector_config.json), [model](https://huggingface.co/deepset/sentence_bert) (the previous model)


Best results were achieved with T-SNE hyperparams: ppl 10-20, LR 0.01, 1-2k steps
It's also better to switch `Label By` to `Title`.

Clearly identifiable clusters
 * Types
 * Completion
 * Search
 * Summarization
 * Bugs (localization, fix/repair)
 * (OpenAI) Commit messages, Decompilation/Obfuscation, ...

Let me know what you think!

**Comment by mallamanis** [comment]
This looks great! Thanks a lot for this 💯

**Comment by mallamanis** [comment]
Hi @bzz it seems like that the Action Fails with this change, I suspect this is due to some restriction on GitHub Actions (memory?). Do you maybe have time to investigate?

https://github.com/ml4code/ml4code.github.io/actions/runs/4952439871

For now, I've reverted this PR in #103

**Comment by bzz** [comment]
Oh, my! From a quick glance - it may have to do with the changes in CI runner image https://github.com/actions/runner-images/discussions/7188 🤔 

I'll setup CI on my fork to try to reproduce and report back.

**Comment by bzz** [comment]
> I suspect this is due to some restriction on GitHub Actions (memory?)

You are right, I missed that the CI has failed repeatedly and it's easily reproducible.

Here is RAM profile across different batch sizes (1 to 512, doubling every ~20sec) 🙄 
all-LM6 mem profile
So the VM gets killed with OOM after exceeding [7Gb RAM limit](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources)
